test/gtest/test_transfer.cpp

```diff
@@ -51,12 +51,46 @@ class MemBuffer : std::shared_ptr<void> {
         return size;
     }
 
+    void fill(uint64_t seed)
+    {
+        for (auto &value : *this) {
```

Comment:
This whole approach of inheriting from a shared_ptr, then extending it with bunch of methods to allow iteration, etc. make the code a bit hard to read and understand. 
Instead, why not make MemBuffer a simple template with single specialization for nixl_mem_t::DRAM_SEG that would just be a wrapper over std::vector<uint64_t> (returning vector's iterators for begin(), end(), etc.) and all methods like fill, check, alignedSize, etc. - local static functions that are not members of the class?

Reply:
yes, the problem here is that the fill/check methods will not be directly useful for GPU memory anyway. will try to refactor.